packages/synapse-react-client/src/utils/jsonschema/SchemaAnnotationUtils.test.ts

```diff
@@ -0,0 +1,93 @@
+import { hasSchema } from '@hyperjump/json-schema'
+import { registerSchema } from '@hyperjump/json-schema/draft-07'
+import { renderHook, waitFor } from '@testing-library/react'
+import { JSONSchema7 } from 'json-schema'
+import { useRegisterSchema } from './SchemaAnnotationUtils'
+
+vi.mock('@hyperjump/json-schema', async () => {
+  const actual = await vi.importActual('@hyperjump/json-schema')
+  return {
+    ...actual,
+    hasSchema: vi.fn(),
+  }
+})
+
+vi.mock('@hyperjump/json-schema/draft-07', async () => {
+  const actual = await vi.importActual('@hyperjump/json-schema/draft-07')
+  return {
+    ...actual,
+    registerSchema: vi.fn(),
+  }
+})
+
+const mockHasSchema = vi.mocked(hasSchema)
+const mockRegisterSchema = vi.mocked(registerSchema)
+
+const MOCK_SCHEMA: JSONSchema7 = {
+  $id: 'https://example.com/test.schema.json',
+  type: 'object',
+  properties: {
+    foo: {
+      type: 'string',
+    },
+  },
+}
+
+describe('useRegisterSchema', () => {
```

Comment:
Tests for an extracted hook to register a schema with the `@hyperjump/json-schema` store.